src/broker/log.c

```diff
@@ -75,11 +76,12 @@ static struct logbuf_entry *logbuf_entry_create (const char *buf, int len)
     struct logbuf_entry *e = calloc (1, sizeof (*e));
     if (!e)
         return NULL;
-    if (!(e->buf = calloc (1, len))) {
+    if (!(e->buf = malloc (len + 1))) {
```

Comment:
Curious, why not keep `calloc (1, len+1)` and then drop the explicit NUL below?
I'm sure you have a good reason, it just caught my eye.

Also a more detailed comment about the purpose of the "stealth" NUL might help. I'm still a bit confused -- this is for ease of decoding the raw payload directly to a nul-terminated string?

Reply:
Since the calloc (1, len) was followed immediately by memcpy (dst, src, len), the zeroing of the buffer seemed like extra work for naught. 

> this is for ease of decoding the raw payload directly to a nul-terminated string

Yes, I should have stated that explicitly.  Will fix.

Reply:
That makes sense. I always assumed that calloc was able to get pre-zeroed memory from the OS e.g. a zero page with COW etc. However, that probably is not the case for small allocations anyway. It also helps to explicitly instead of implicitly show the string termination.

Reply:
I think [you're right](https://sources.debian.org/src/glibc/2.28-10/malloc/malloc.c/#L3452) about the calloc implementation.